stdlib/cmake/modules/AddSwiftStdlib.cmake

```diff
@@ -94,12 +94,9 @@ function(_add_target_variant_c_compile_link_flags)
   endif()
 
   if("${CFLAGS_SDK}" STREQUAL "ANDROID")
-    # lld can handle targeting the android build.  However, if lld is not
-    # enabled, then fallback to the linker included in the android NDK.
-    if(NOT SWIFT_USE_LINKER STREQUAL "lld")
-      swift_android_tools_path(${CFLAGS_ARCH} tools_path)
-      list(APPEND result "-B" "${tools_path}")
-    endif()
+    # Make sure the Android NDK lld is used.
+    swift_android_tools_path(${CFLAGS_ARCH} tools_path)
+    list(APPEND result "-B" "${tools_path}")
```

Comment:
Brought the NDK tools path back here and tested it by passing the prebuilt clang that comes with the Swift trunk snapshot toolchain, as that's the clang a fresh build would use too, to `--native-clang-tools-path` when cross-compiling the standalone stdlib, only to hit another problem: it's trying to link against libgcc. I dug into this difference and [the NDK clang hardcodes using compiler-rt instead](https://android.googlesource.com/toolchain/llvm_android/+/refs/heads/master/patches/cherry/a478b0a199f4928041390d1f35dd5e226936b42c.patch#30), along with [other changes related to libunwind](https://android.googlesource.com/toolchain/llvm_android/+/refs/heads/master/patches/cherry/91f8aacc040f5a458cfc2a10c4039e0028931455.patch).

We may be better off changing the clang used to link for Android with a freshly-built Swift toolchain to the NDK clang too, let me know what you think.

Reply:
@drodriguez, just waiting to hear from you about this.

Reply:
Sorry for the radio silence. It has been a busy couple of weeks lately.

What I have done:

- Upgraded the CI machines to Ubuntu 18.04. This was required with the recent drop of support for 16.04, which this machines were using.
- Checked the rebranch that was supposed to happen soon.
- NDK r23 was installed on the machines. Not being used, but ready to be used.
- All the Termux libraries that you required as dependencies are installed in two directories, ready to be used. They are all installed as if they were in Termux, which makes using them a question of adding an include and a library path, which should simplify things (I hope).

What I have **not** done:

- Checked the rebranch recently.
- Checked that this changes work on CI.
- Checked if Termux ICU can replace the aging ICU libraries that currently CI uses.

I will probably try to find out some time for the rebranch check (because the clock is ticking on that one). I will try to get to the others as soon as I can.

Reply:
I've been getting [the stdlib built for Android with the Swift-forked clang](https://github.com/buttaface/swift-android-sdk/pull/19/files#diff-76992bc322b65ab94f299c05ba3cbd8b3ab2e32ebfdde6db5adba70fa9f5a26aR132) and I just noticed that clang 13 switches Android to building with compiler-rt and libunwind, llvm/llvm-project@91f8aacc0, which was upstreamed from the NDK clang. Since the upcoming Swift rebranch has that patch too, I think it will break the community Android CI, unless we switch it to Android NDK 23.

We should probably get this NDK 23 pull in before the rebranch: the good news is that I have the Swift-forked clang building the stdlib for Android now, with the patches in my first link. I've just kicked off a build to make sure the validation suite passes for me locally on linux, just as the Android CI runs with `--host-test --skip-test-linux`, and will update this pull with those changes once that passes. You can push the work with Termux packages till we get the corelibs built with #38441, as those Android library packages are only really needed for that pull, not this one.